statsmodels/regression/linear_model.py

```diff
@@ -818,21 +822,42 @@ def iterative_fit(self, maxiter=3):
         ----------
         maxiter : integer, optional
             the number of iterations
+        rtol : float, optional
+            Relative tolerance between estimated coefficients to stop the
+            estimation.  Stops if
+
+            max(abs(last - current) / abs(last)) < rtol
+
         """
-        #TODO: update this after going through example.
-        for i in range(maxiter-1):
+        # TODO: update this after going through example.
+        for i in range(maxiter - 1):
             if hasattr(self, 'pinv_wexog'):
                 del self.pinv_wexog
             self.initialize()
             results = self.fit()
+            self.history.append(results)
+            if i == 0:
+                last = results.params
+            else:
+                diff = np.max(np.abs(last - results.params) / np.abs(last))
+                if diff < rtol:
+                    break
+                last = results.params
             self.rho, _ = yule_walker(results.resid,
                                       order=self.order, df=None)
-        #why not another call to self.initialize
+        # why not another call to self.initialize
         if hasattr(self, 'pinv_wexog'):
             del self.pinv_wexog
         self.initialize()
-        results = self.fit() #final estimate
-        return results # add missing return
+        if _is_using_pandas(self.history[0], None):
+            history = pd.concat(history, 1)
+            history.columns = ['iter_' + int(i) for i in range(history.shape[1])]
+        else:
+            history = np.column_stack(self.history)
+        # Use kwarg to insert history
+        results = self.fit(history=history)
```

Comment:
so far we patch the results instance directly by assigning directly the attribute
`results.history = history`

That's up for review and being changed since just assigning attributes outside of `__init__` is not really "nice", but having `history` as a fit keyword also feels strange.  (I did both, using kwargs and direct assignment, last time I changed something similar.)